prov/sockets/src/sock_fabric.c

```diff
@@ -634,6 +622,57 @@ int sock_node_getinfo(const char *node, const char *service, uint64_t flags,
 	return ret;
 }
 
+static int sock_match_src_addr(struct slist_entry *entry, const void *src_addr)
+{
+	struct sock_host_list_entry *host_entry;
+	host_entry = container_of(entry, struct sock_host_list_entry, entry);
+
+        return ((strcmp(host_entry->hostname, (char *) src_addr) == 0) ||
+		(strcmp(host_entry->hostname, "0.0.0.0") == 0));
+}
+
+static int sock_matches_interface(struct slist addr_list, const void *src_addr,
+				   size_t src_type_len)
```

Comment:
As Jithin mentioned below, using a len to indicate the address type is confusing.  Given the structure of the function and that it nearly splits into 2 based on src_type_len, I would just create 2 separate functions and call the correct one from the proper location.